src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageAnalyzerResultProvider.cs

```diff
@@ -115,8 +120,10 @@ void GetResults<T>(string name, string action, IReadOnlyCollection<Operation<T>>
                             {
                                 results.UnionWith(s.OperationDetails.Details.Select(s => new AnalyzeResult()
                                 {
-                                    RuleId = Id,
-                                    RuleName = this.Name,
+                                    RuleId = RuleId,
+                                    RuleName = RuleName,
+                                    FullDescription = FullDescription,
+                                    HelpUri = _helpUri,
```

Comment:
Maybe we can split each of these operations into a separate rule, that way they will be grouped under the category. For eg: `Packages to be added` is a rule, `Packages to be deleted` is a separate rule so on and so forth. Is that doable here?

Reply:
That way each of these rules will have their own custom Rule id, name and description.

Reply:
SARIF contains one thing that we call SubId.
RuleId = RULE0001
SubId = Rule0001/Update
SubId = Rule0001/Delete
SubId = Rule0001/Create

The same RuleId contains 3 specific formats (update, create, and delete). This is useful when people wants to suppress the specific format (for example, only suppressing the delete option).

If you have 3 different guidance, then we should create 3 separate rules (rule0001, rule0002, rule0003). If all will point to the same guidance, then, we could create just one rule.

Reply:
SubId seems like right choice here, is it possible to make the change here for this PR? @eddynaka . Lets sync offline to design this a little better.

Reply:
We synced offline and are going with separate ruleids here